tests/PHPStan/Rules/Comparison/MatchExpressionRuleTest.php

```diff
@@ -221,7 +221,12 @@ public function testBug7622(): void
 			$this->markTestSkipped('Test requires PHP 8.0.');
 		}
 		$this->treatPhpDocTypesAsCertain = false;
-		$this->analyse([__DIR__ . '/data/bug-7622.php'], []);
+		$this->analyse([__DIR__ . '/data/bug-7622.php'], [
+			[
+				'Match expression does not handle remaining value: string',
+				21,
```

Comment:
This is wrong

Reply:
Is this? I'm not sure if native type can know that the match is exhaustive.

Reply:
But the rule always asks getType, not getNativeType. The pupose of the setting `treat...` to false is to remove errors like "this if is always false" (when code checks again something that's only expressed in PHPDocs and not checked at runtime), not add new errors.

Reply:
ok, I think
> The pupose of the setting treat... to false is to remove errors like "this if is always false" 

was what I was totally misunderstanding.

My thinking was to switch between `native types` and `full types` by treatPhpDocTypesAsCertain

like,
`treatPhpDocTypesAsCertain: true`
getNativeType() => native type
getType() => full types

`treatPhpDocTypesAsCertain: false`
getNativeType() = getType() => native type

I now understood that when a rule error is
|-|full type|native type|
|-|-|-|
|1|no error|no error|
|2|error|no error|
|3|no error| error|
|4|error| error|

purpose of `treatPhpDocTypesAsCertain` is to relax cases like 2.

Reply:
I think the docs are clear on this 😊 "relaxes" https://phpstan.org/config-reference#treatphpdoctypesascertain